aspnetcore/grpc/loadbalancing.md

```diff
@@ -0,0 +1,99 @@
+
+* **Client-side load balancing** - The gRPC client will make a load balancing decision when a gRPC call is started. The gRPC call is sent directly to the final endpoint. In this model, the client is responsible for knowing about available endpoints. Client-side load balancing is a high-performance solution that eliminates the need for a proxy and additional network hops.
+
+## Configure gRPC client-side load balancing
+
+Client-side load balancing is configured when a channel is created. The two components to consider when using load balancing are the address resolver, which will resolver the addresses; and the load balancer, which will pick which address should be used for a gRPC call.
+
+The address resolver is configured using the scheme of the address URI. For example, a `dns` scheme maps to `DnsAddressResolver`. This resolver will get the Internet Protocol (IP) addresses for the specified address.
+
+```csharp
+var channel = GrpcChannel.ForAddress(
+    "dns://backend.default.svc.cluster.local",
+    new GrpcChannelOptions { ChannelCredentials = ChannelCredentials.Insecure });
+var client = new Greet.GreeterClient(channel);
+
+var response = await client.SayHelloAsync(new HelloRequest { Name = "world" });
+```
+
+The preceding code:
+
+* Configures the created channel with the address `dns://backend.default.svc.cluster.local`. The `dns` schema maps to `DnsAddressResolver`.
+* Doesn't specify a load balancer. The channel will default to `PickFirstLoadBalancer`.
+* Starts the gRPC call `SayHello`.
+  * `DnsAddressResolver` will resolve IP addresses for the host name `backend.default.svc.cluster.local`. The result is cached and periodically refreshed. The host name value will depend upon your environment.
+  * `PickFirstLoadBalancer` will attempt to connect to one of the resolved IP addresses.
+  * The call will be sent to the first IP address the channel successfully connects to.
+
+A load balancer is specified in the service config. If no service config or load balancer is specified then the channel will default to `PickFirstLoadBalancer`. Another load balancer implementation that is included is `RoundRobinLoadBalancer`. This load balancer will attempt to connect to all IP addresses returned by the address resolver. The load balancer then distributes load across all IP addresses.
+
+```csharp
+var channel = GrpcChannel.ForAddress(
+    "dns://custom-hostname",
+    new GrpcChannelOptions
+    {
+        ChannelCredentials = ChannelCredentials.Insecure,
+        ServiceConfig = new ServiceConfig { LoadBalancingConfigs = { new RoundRobinConfig() } }
+    });
+var client = new Greet.GreeterClient(channel);
+
+var response = await client.SayHelloAsync(new HelloRequest { Name = "world" });
+```
+
+The preceding code:
+
+* Specifies a round robin load balancer in the service config.
```

Comment:
```suggestion
* Specifies a round robin load balancer in the service configuration.
```

Reply:
"service config" is an actual gRPC name for the referred collection of settings. I'd like to leave it as is.